recipes/lerc/all/conanfile.py

```diff
@@ -73,6 +73,8 @@ def package_info(self):
         self.cpp_info.libs = ["LercLib" if Version(self.version) < "4.0.0" else "Lerc"]
         if self.settings.os in ["Linux", "FreeBSD"]:
             self.cpp_info.system_libs.append("m")
+        if stdcpp_library(self):
```

Comment:
Why?

Reply:
@uilianries I'm genuinely interested in the reasoning so I can take this into account in the future.

The shared build indeed does not strictly need the C++ stdlib dependency thanks to the `.dynamic` section, but I think this info is still valuable for the case of otherwise pure-C applications being packaged/deployed correctly.

Reply:
I'll ping @jcar87 as he has a detailed insight about the case.

Reply:
Shared libraries contain metadata that specifies their own dependencies, and the dynamic linker uses this information to automatically load the necessary shared library. The same does not happen when linking to static libraries, that metadata is not embedded, so you need to pass dependencies to the linker. That's why you only need to specify to link to stdcpp library when using static.